src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -162,7 +161,29 @@ private CachedCoinView(IDateTimeProvider dateTimeProvider, ILoggerFactory logger
         }
 
         /// <inheritdoc />
-        public override async Task<FetchCoinsResponse> FetchCoinsAsync(uint256[] txIds, CancellationToken cancellationToken = default(CancellationToken))
+        public Task<uint256> GetTipHashAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Task<uint256> task = Task.Run(async () =>
+            {
+                this.logger.LogTrace("()");
+
+                if (this.blockHash == null)
+                {
+                    FetchCoinsResponse response = await this.FetchCoinsAsync(new uint256[0], cancellationToken).ConfigureAwait(false);
```

Comment:
Old method I didn't change. Only implement method for getting tip for those that relied on a method that used to be in abstract class and not it's interface.

Reply:
Probably in one of next steps I'll include those changes. 
I tried to keep this PR small because it already touches a lot of files just by adding an interface.

Reply:
I just don't understand that description of the PR.